client/src/components/RepositoryItem/index.jsx

```diff
@@ -34,19 +34,20 @@ class RepositoryItem extends React.Component {
     await this.checkIfEmpty(username, name);
 
     const { isEmpty } = this.state;
-    const { repoBranches } = this.state;
-    if (!isEmpty || (repoBranches && repoBranches.length)) {
+
+    if (!isEmpty) {
       try {
         await this.getRepoBranches(repoId);
+        const { repoBranches } = this.state;
 
         let allRepoCommits = [];
         repoBranches.forEach(async branch => {
           const allBranchCommits = this.getRepoCommits(repoId, branch.name);
           allRepoCommits.push(allBranchCommits);
         });
         Promise.all(allRepoCommits).then(data => {
-          const allRepoCommitsSorted = data.flat().sort((a, b) => new Date(b.date) - new Date(a.date));
-          const lastCommitDate = moment(allRepoCommitsSorted[0].date).fromNow();
+          const allRepoCommitsSorted = data.flat().sort((a, b) => new Date(b.updatedAt) - new Date(a.updatedAt));
```

Comment:
У нас точная дата создания коммита находится в createdAt, и коммит нигде не апдейтится 
```suggestion
          const allRepoCommitsSorted = data.flat().sort((a, b) => new Date(b.createdAt) - new Date(a.createdAt));
```